app/models/institution_builder.rb

```diff
@@ -453,21 +454,22 @@ def self.add_yellow_ribbon_programs(version_number)
         (SELECT
           i.version, i.id, yrs.degree_level, yrs.division_professional_school,
             yrs.number_of_students, yrs.contribution_amount, NOW(), NOW()
-          FROM institutions as i, yellow_ribbon_program_sources as yrs
+          FROM institutions as i, yellow_ribbon_program_sources as yrs, versions
           WHERE i.facility_code = yrs.facility_code
-          AND i.version = ?);
+          AND versions.number = #{version_number}
+          AND versions.id = i.version_id);
```

Comment:
I think this can also be reduced to 
```
          FROM institutions as i, yellow_ribbon_program_sources as yrs
          WHERE i.facility_code = yrs.facility_code
          AND i.version_id = #{version_id};
```

Maybe in a separate PR, but `YellowRibbonProgram` looks like a candidate to reduce/remove it's `version` attribute since it `belongs_to` `institution`.
You'll have to create a migration to stop requiring it be populated:
```
  def change
    change_column_null :yellow_ribbon_programs, :version, true
  end
```
Then you can stop populating it, and eventually remove it.

Reply:
I had actually made this refactor, but it caused the majority of the `institution_builder_spec` queries to bomb out, so I left it out of the refactor for now. 

I also noticed `YellowRibbonProgram` here using the `version` attribute and had similar feelings regarding that, but I think that would probably be best handled in a separate PR.